File: 3163-string-compression-iii/3163-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string result = "";
        int cnt = 1;

        for (int i = 1; i < word.length(); i++) {
            if (word[i] == word[i - 1]) {
                if (cnt == 9) {
                    result += to_string(cnt);
                    result += word[i - 1];
                    cnt = 1;
                } else {
                    cnt++;
                }
            } else {
                result += to_string(cnt);
                result += word[i - 1];
                cnt = 1;
            }
        }

        result += to_string(cnt);
        result += word[word.length() - 1];

        return result;
    }
};

File: 3163-string-compression-iii/3163-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder result = new StringBuilder();
        int cnt = 1;

        for (int i = 1; i < word.length(); i++) {
            if (word.charAt(i) == word.charAt(i - 1)) {
                if (cnt == 9) {
                    result.append(cnt);
                    result.append(word.charAt(i - 1));
                    cnt = 1;
                } else {
                    cnt++;
                }
            } else {
                result.append(cnt);
                result.append(word.charAt(i - 1));
                cnt = 1;
            }
        }

        result.append(cnt);
        result.append(word.charAt(word.length() - 1));

        return result.toString();
    }
}